src/seq_mv/csr_matop_device.c

```diff
@@ -1322,12 +1343,9 @@ hypre_CSRMatrixTriLowerUpperSolveCusparse(char             uplo,
       hypre_CSRMatrixData(A) = A_a;
 #endif
 
-      hypre_SortCSRCusparse(nrow, ncol, nnzA, A_i, A_sj, A_sa);
+      hypre_SortCSRCusparse(nrow, ncol, nnzA, descr, A_i, A_sj, A_sa);
    }
 
-   cusparseHandle_t handle = hypre_HandleCusparseHandle(hypre_handle());
-   cusparseMatDescr_t descr = hypre_HandleCusparseMatDescr(hypre_handle());
-
    HYPRE_CUSPARSE_CALL( cusparseSetMatDiagType(descr, CUSPARSE_DIAG_TYPE_NON_UNIT) );
```

Comment:
`descr` of A is changed here (FILL_MODE, UNIT DIAG). But it seems that cusparse spmv doesn't pick them up. Not sure if rocsparse will do.

Reply:
The APIs exist for doing this in rocSPARSE. But I'm unsure exactly what they affect.